frontend/src/scenes/cohorts/cohortLogic.ts

```diff
@@ -113,17 +119,82 @@ export const cohortLogic = kea<cohortLogicType<CohortLogicProps>>({
                 },
             },
         ],
+        pollTimeout: [
+            null as NodeJS.Timeout | null,
+            {
+                setPollTimeout: (_, { pollTimeout }) => pollTimeout,
+            },
+        ],
         lastSavedAt: [
             false as string | false,
             {
                 setLastSavedAt: (_, { lastSavedAt }) => lastSavedAt,
             },
         ],
-        submitted: [
-            // Indicates the form has been submitted at least once. Used to display validation errors if applicable.
-            false,
+    }),
+
+    forms: ({ actions }) => ({
+        cohort: {
+            defaults: processCohortOnSet(NEW_COHORT),
+            validator: ({ name, csv, is_static, id, groups }) => ({
```

Comment:
It'd be great to have some tests for this logic. The validator seems like it could use some, but it generally feels like there's some good logic in this file to be tested.